### ISSUE TITLE
error encountered with textext 0.7.3 and 0.8 using TeXLive 2018

### ISSUE THREAD

**Issue by yuxinopt** [textext/textext#34]
Upon Preview I received error:
[Error 145] : 'c:\\users\\admini~1\\appdata\\local\\temp\\tmp9etrww'

Upon Save I received error:

[Error 145] : 'c:\\users\\admini~1\\appdata\\local\\temp\\tmphc_sfz'
Traceback (most recent call last):
  File "asktext.py", line 583, in cb_ok
    self._texcmd_cbox.get_active_text().lower())
  File "textext.py", line 286, in callback
    tex_cmd, original_scale=current_scale)
  File "textext.py", line 389, in do_convert
    converter.finish()
  File "textext.py", line 703, in finish
    self.remove_temp_files()
  File "textext.py", line 802, in remove_temp_files
    self.try_remove(self.tmp_path)
  File "textext.py", line 810, in try_remove
    os.rmdir(filename)
WindowsError: [Error 145] : 'c:\\users\\admini~1\\appdata\\local\\temp\\tmphc_sfz'
The same errors occur for both textext0.7.3 and 0.8. I am under Windows 7.

**Comment by jcwinkler** [comment]
The deletion of the temporary directory used for compilation of the TeX code fails. 

It would be great if you could help us by performing the following test:

1. Open the file `textext.py` (it should be in the directory `c:\Users\admini1~1\AppData\Roaming\inkscape\extensions` in your case)

2. Navigate to the line containing `def try_remove(filename):` (it should be line 805 in version 0.8)

3. Modify the code that it looks like this:

```python
    def try_remove(filename):
        """Try to remove given file, skipping if it doesn't exist"""
        if os.path.isfile(filename):
            inkex.debug(filename) # <-- added
            os.remove(filename)
        elif os.path.isdir(filename):
            inkex.debug(os.listdir(filename)) # <-added
            os.rmdir(filename)
```

4. Save the file

5. Try again to compile a TeX snippet

6. After the error message a window like this should open:
![grafik](https://user-images.githubusercontent.com/11866252/40776857-49e47e24-64cc-11e8-98cd-ed62a57d69c1.png)

7. Please post its content here.

Thank you very much!

**Comment by yuxinopt** [comment]
Thanks.

I did the test following your guide. Errors remain the same, but no such window pops out.

Best wishes

**Comment by jcwinkler** [comment]
Meanwhile I was able to reproduce the problem, see commit 76aab088b3e81bc2b74d96ed9fafa231133030c5

It would be nice if you could check this on your system. Please perform the following steps:
1. Close Inkscape
2. Open your Inkscape extension directory (should be `c:\Users\admini1~1\AppData\Roaming\inkscape\extensions`)
3. Remove the files `textext.inx` **Note**: The **inx** file, not py !!
4. Open Inkscape again and check if the TexText extension **does not** show up in the Extension menu. If it shows up, please inform me
5. Close Inkscape again.
6. Download the attached zip file [textext-tempdir-fix.zip](https://github.com/textext/textext/files/2064156/textext-tempdir-fix.zip) and unpack its content into the Inkscape extension directory from step 2, overwrite all files
7. Launch Inkscape and try out TexText again. What happens?

Thank you very much in advance!

**Comment by yuxinopt** [comment]
jcwinkler, thank you very much!

I did the check, the results are:

1. Upon removing the textext.inx file, the TeXText extension does not show up in the extension menu.

2. Upon using the fix.zip files, I got the following errors:
upon preview:  [Error 2] : 'texput.log'

upon save: 

[Error 2] : 'texput.log'
Traceback (most recent call last):
  File "asktext.py", line 583, in cb_ok
    self._texcmd_cbox.get_active_text().lower())
  File "textext.py", line 286, in callback
    tex_cmd, original_scale=current_scale)
  File "textext.py", line 389, in do_convert
    converter.finish()
  File "textext.py", line 703, in finish
    self.remove_temp_files()
  File "textext.py", line 802, in remove_temp_files
    self.try_remove(self.tmp_path)
  File "textext.py", line 815, in try_remove
    os.rmdir(filename)
WindowsError: [Error 2] : 'texput.log'


Best regards

**Comment by jcwinkler** [comment]
Hmm, I am quite puzzled about what's going on on your system. I use Windows 7 and TexLive too as a test system...

Please perform the following test:
1. Close Inkscape
2. Download the attached zip file [textext-no-temp-delete.zip](https://github.com/textext/textext/files/2069579/textext-no-temp-delete.zip) and unpack its content into the Inkscape extension directory (`c:\Users\admini1~1\AppData\Roaming\inkscape\extensions`), overwrite the existing textext.py
3. Launch Inkscape and try out TexText again. This time a window as described in my first post should appear after hitting "Save". Please post its content.

In the attached file I completely removed the deletion of the temp directory created by TexText so there sould nothing go wrong. Instead the files and dirs TexText wants to delete are displayed only in the window.

Thank you very much.

**Comment by yuxinopt** [comment]
Thank you very much for the efforts! In fact I did another test. I rolled back from TeXLive 2018 to TeXLive 2016 and all the errors disappeared.

Best wishes

**Comment by jcwinkler** [comment]
Although not directly verified by the user the reason for this issue was a non safe deletion of the temp directory which has been fixed by commit 8c65ec0 and which went into release 0.8 of TexText.
